tests/semantic/scripts/expressions.py

```diff
@@ -28,3 +28,5 @@
 
 # this statement will be ignored at the codegen
 x = ad is None
```

Comment:
Is this comment still correct? Shouldn't `x` just be a boolean variable, and more specifically `x := False`?

Reply:
I don't think this statement is entirely correct anymore. It might be good to implement this behaviour.
Currently the presence of `ad is None` is used to mark the variable as optional. If the user writes:
`if (ad is None):`
Then this is interpreted as `If (.not. present(ad))`
However the assignment case hasn't been handled and so a warning still appears and the line still isn't printed.
This is a quick fix, however a program is not the correct place to test whether it works.
I don't know whether it should have different behaviour in a program or if we should just leave it as is as there is no reason to write this in a program unless you are using None as a variable, which cannot be translated anyway

Reply:
I see. This should work inside of a function with optional arguments, but it makes no sense in a program. Thanks for looking into that.